src/Compilers/CSharp/Test/Syntax/Parsing/RoleParsingTests.cs

```diff
@@ -1313,7 +2408,1184 @@ public void RoleParsing_WithoutBody_WithoutBaseList_WithTypeParameter(bool isExt
+                            }
+                        }
+                        N(SyntaxKind.DotToken);
+                    }
+                    N(SyntaxKind.OperatorKeyword);
+                    N(SyntaxKind.PlusToken);
+                    N(SyntaxKind.ParameterList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.Parameter);
+                        {
+                            N(SyntaxKind.QualifiedName);
+                            {
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "extension");
+                                }
+                                N(SyntaxKind.DotToken);
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "I");
+                                }
+                            }
+                            N(SyntaxKind.IdentifierToken, "one");
+                        }
+                        N(SyntaxKind.CommaToken);
+                        N(SyntaxKind.Parameter);
+                        {
+                            N(SyntaxKind.QualifiedName);
+                            {
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "extension");
+                                }
+                                N(SyntaxKind.DotToken);
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "I");
+                                }
+                            }
+                            N(SyntaxKind.IdentifierToken, "other");
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                    N(SyntaxKind.ArrowExpressionClause);
+                    {
+                        N(SyntaxKind.EqualsGreaterThanToken);
+                        N(SyntaxKind.ThrowExpression);
+                        {
+                            N(SyntaxKind.ThrowKeyword);
+                            N(SyntaxKind.NullLiteralExpression);
+                            {
+                                N(SyntaxKind.NullKeyword);
+                            }
+                        }
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.CloseBraceToken);
+            }
+            N(SyntaxKind.EndOfFileToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void ExplicitConversionOperator_FromExtensionType()
+    {
+        var text = """
+class C
+{
+    public static explicit operator C(extension i) { throw null; }
```

Comment:
Sorry, this is still not what I'm asking for. I'm asking for an implicit and explicit conversion, explicitly implemented, from a type named extension and from a type in a namespace named extension.

```cs
    static explicit extension<extension>.operator extension(int i)
    static explicit extension.C<C>.operator C(int i)
```

Reply:
you are bad, and you should feel bad :D.

Ugh....  So, i'm ok with a convo about a breaking this case (outside of this PR).  Seems awful to try to continue supporting that :D

Reply:
I'm going to leave this as a prototype comment. Not worth delaying the core of the feature. We can discuss later.

Reply:
FWIW, I could not find a scenario where the above code is legal from a binding standpoint. Can you see one?

Reply:
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAJgjAFMBnABwgGNSACUgDwBdSZyoB7GAbgILoAzPVhsATgDNaDAJIAeACoA+egHcAFqXENF9EPQUqCAbwL0L9YfQjByLcbRZXsANkZNKAGyg0ozrkptCBYucVEYFgAKPSgASn58AF8BfCx6AGF9Q3kM5VNzS3Q3D29ffw82Dm4YADoFPNrA4NDwsSis+MKLM3xLfqsAdnpsRP6U5KA===.